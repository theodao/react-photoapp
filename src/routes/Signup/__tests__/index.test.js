import React from 'react';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Signup, mapStateToProps, mapDispatchToProps } from '../';

describe('Testing sign up route', () => {
  afterEach(cleanup);

  const pushFunction = jest.fn();

  const history = {
    push: pushFunction,
  };

  beforeAll(() =>
    jest.spyOn(React, 'useEffect').mockImplementation(React.useLayoutEffect),
  );
  afterAll(() => React.useEffect.mockRestore());

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <Signup
        auth={{
          isLoggedIn: false,
        }}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should navigate back to dashboard when user is not logged in', () => {
    const tree = renderer.create(
      <Signup
        auth={{
          isLoggedIn: true,
        }}
        history={history}
      />,
    );
    tree.update();
    expect(pushFunction).toHaveBeenCalled();
  });

  it('Testing mapStateToProps', () => {
    const state = {
      auth: {
        isLoggedIn: false,
      },
    };

    const transformedState = mapStateToProps(state);

    expect(transformedState.auth.isLoggedIn).toBe(false);
  });

  it('Testing mapDistpachToProps', () => {
    const mockedDispatchFunction = jest.fn();

    const dispatchFunctions = mapDispatchToProps(mockedDispatchFunction);

    dispatchFunctions.dispatchSignup();

    expect(mockedDispatchFunction).toHaveBeenCalled();
  });
});
