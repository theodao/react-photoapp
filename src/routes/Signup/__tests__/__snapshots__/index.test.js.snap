// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing sign up route Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ oxXIZ"
  >
    <div
      class="sc-AxiKw EEztA"
    >
      <div
        class="sc-AxhCb sc-AxhUy cRUfQ"
      >
        Create an account
      </div>
      <form>
        <div>
          <input
            class="FormInput u-block u-widthFull u-webkitScrollbar"
            name="username"
            placeholder="Username"
            required=""
            style="font-size: 0.9rem;"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-AxjAm cExXRh"
        />
        <div>
          <input
            class="FormInput u-block u-widthFull u-webkitScrollbar"
            name="email"
            placeholder="Email"
            required=""
            style="font-size: 0.9rem;"
            type="text"
            value=""
          />
        </div>
        <div
          class="sc-AxjAm cExXRh"
        />
        <div>
          <input
            class="FormInput u-block u-widthFull u-webkitScrollbar"
            name="password"
            placeholder="Password"
            required=""
            style="font-size: 0.9rem;"
            type="password"
            value=""
          />
        </div>
        <div
          class="sc-AxjAm cExXRh"
        />
        <button
          class="Button u-flexInline u-justifyContentCenter u-alignItemsCenter u-textDecorationNone u-roundedMedium u-textWhite hover:u-textWhite u-backgroundPrimary hover:u-backgroundPrimaryDark u-border u-borderPrimary hover:u-textDecorationNone is-disabled u-cursorNotAllow u-pointerEventsNone u-widthFull u-textUppercase"
          disabled=""
          style="font-size: 0.9rem; opacity: 1;"
        >
          <span
            class="Button-label u-inlineBlock"
          >
            Sign up
          </span>
        </button>
      </form>
      <div
        class="sc-AxhCb jZbWFZ u-cursorPointer"
      >
        Log In instead ?
      </div>
    </div>
  </div>
</DocumentFragment>
`;
